contributing.md

```diff
@@ -1,65 +1,78 @@
 
+---
 
-<h1 align="center">Creating your First Pull Request 🌟</h1>
+## Prerequisites 📋
 
----
+Before getting started, ensure you have the following tools installed:
+
+- **Node.js** (version >= 14.x)
+- **npm** (version >= 6.x)
+
+You can check if Node.js and npm are installed by running:
+```bash
+node -v
+npm -v
+```
+## Creating Your First Pull Request 🌟
+## 1. Star this repository: Show some love by starring the repository!
+## 2. Fork this repository: Click the 'Fork' button at the top right of this page to create a copy of this repository in your account.
+## 3. Clone the forked repository:
 
-1. Star this repository.
-2. Fork this repository.
-3. Clone the forked repository.
 ```bash
 git clone https://github.com/<your-github-username>/ElectroKart.git
 ```
 
-4. Navigate to the project directory.
+## 4. Navigate to the project directory
+
 ```bash
 cd ElectroKart
 ```
-5. Install the dependencies.
+## 5. Install the dependencies:
 ```bash
 npm install
 ```
-6. Create a new branch.
+## 6. Create a new branch (use descriptive branch names
+   feature/<feature_name> or fix/<issue_name>):
 ```bash
-git checkout -b <your_branch_name>
+git checkout -b feature/<your_branch_name>
 ```
-7. Make changes and ensure the app runs correctly.
+
+## 7. Make changes and ensure the app runs correctly:
 ```bash
 npm start
 ```
-8. Stage your changes and commit.
+## 8. Stage your changes and commit (use descriptive commit messages like
+   feat: added new feature or fix: corrected issue):
 ```bash
 git add .
+git commit -m "feat: added new feature"
 ```
-```css
-
-git commit -m "Added new feature/fixed bug."
-```
-
-9. Push your local commits to the remote repo.
+## 9. Push your local commits to the remote repository:
 ```bash
-git push -u origin <your_branch_name>
+## git push -u origin feature/<your_branch_name>
 ```
+## 10. Create a Pull Request (PR):
+    Provide a detailed description of the changes you've made.
+    If it's a UI change, consider adding screenshots for better understanding.
+## 11. Congratulations! 🎉 You've made your contribution.
 
-10. Create a Pull Request.
-11. Congratulations! 🎉 you've made your contribution.
-
-
-## :zap: Pull Requests Review Criteria 🧲 
+## Pull Request Review Criteria 🧲
 
-1. Please fill the ***PR Template*** properly while making a Pull Request.
-2. Never commit in the `main` branch.
-3. Your work must be original, written by you not copied from other resources.
-4. You must comment on your code where necessary.
+1. Please fill out the PR template properly when creating a pull request.
+2. Assign yourself to the issue you’re working on to help us track progress.
+3. Never commit to the main branch.
+4. Your work must be original and adhere to the project's coding standards.
+5. Comment your code where necessary for clarity.
+6. Always ensure all tests pass before pushing your changes.
```

Comment:
great job  @rajatsinghal02 , on the updates! Please add “Always ensure all tests pass before pushing your changes by running: npm run test” to enhance clarity for new contributors. Thank you!